src/main/java/io/jenkins/plugins/services/impl/DefaultConfigurationService.java

```diff
@@ -123,48 +134,97 @@ protected String getJiraPassword() {
     return StringUtils.trimToNull(System.getenv("JIRA_PASSWORD"));
   }
 
-  public CredentialsProvider getJiraCredentials() {
-    CredentialsProvider credsProvider = new BasicCredentialsProvider();
-    credsProvider.setCredentials(
-      AuthScope.ANY,
-      new UsernamePasswordCredentials(this.getJiraUsername(), this.getJiraPassword()));
-    return credsProvider;
+  public List<Header> getJiraCredentials() {
+    String encoding = Base64.getEncoder().encodeToString((this.getJiraUsername() + ":" + this.getJiraPassword()).getBytes());
+    return Collections.singletonList(new BasicHeader(HttpHeaders.AUTHORIZATION, "Basic " + encoding));
   }
 
-  public static String _getGithubClientId() {
-    String clientId = StringUtils.trimToNull(System.getenv("GITHUB_CLIENT_ID"));
-    if (clientId != null) {
-      return clientId;
+  // borrowed from https://github.com/jenkinsci/github-branch-source-plugin/blob/07b69ab4f9d9ed718416ac63af67102de5e172fa/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java#L107
+  @SuppressWarnings("deprecation") // preview features are required for GitHub app integration, GitHub api adds deprecated to all preview methods
+  String generateAppInstallationToken(String appId, String appPrivateKey) {
+    try {
+      String jwtToken = createJWT(appId, appPrivateKey);
+      GitHub gitHubApp = new GitHubBuilder().withEndpoint(this.getGithubApiBase())
+        .withJwtToken(jwtToken)
+        .build();
+
+      GHApp app = gitHubApp.getApp();
+
+      List<GHAppInstallation> appInstallations = app.listInstallations().asList();
+      if (appInstallations.isEmpty()) {
+        throw new IllegalArgumentException(String.format("Couldn't authenticate with GitHub app ID %s", appId));
+      }
+      GHAppInstallation appInstallation;
+      if (appInstallations.size() == 1) {
+        appInstallation = appInstallations.get(0);
+      } else {
+        appInstallation = appInstallations.stream()
+          // .filter(installation -> installation.getAccount().getLogin().equals(owner))
```

Comment:
either

```suggestion
```

or

add a configuration for it

Reply:
ack i didn't push that update, one sec